### PR TITLE
Edit Product: enable editing basic fields for non-core products

### LINKED ISSUE #1998: Edit Product: enable editing basic fields for non-core products
As discussed internally, we need to enable editing product title and description for all product types. Currently, editing is only enabled for Simple products.

Based on the discussion (CGPNUU63E/p1596617913064000?thread_ts=1596591699.062300-slack), we will support the following fields for products that don't have a core product type (simple/variable/external/grouped):

- Name
- Description
- Images
- Price settings (readonly)
- Inventory (readonly)
- Short description
- Categories
- Tags

And all of the product settings.

After this issue, we can delete the old `ProductDetailsViewController` and related code!

### PR DESCRIPTION
**Pull request by jaclync**
Fixes #1998 

## Changes

- Added 2 new cases to `ProductFormEditAction` enum: `readonlyPriceSettings` and `readonlyInventorySettings` (I considered adding an associated value `isEditable` to the existing `priceSettings` and `inventorySettings`, but [we don't have an easy way to provide the default value until we use Swift 5.1](https://www.swiftbysundell.com/tips/default-enum-associated-values/) so I created new enums so the PR isn't huge from passing `isEditable: true` for the editable cases)
- Implemented the setting row view model for the new enum cases in `DefaultProductFormTableViewModel`, passing `isEditable` boolean to `ProductFormSection.SettingsRow.ViewModel`'s `isActionable` field (showing/hiding the disclosure indicator in the accessory view)
- Added `isEditable: Bool` associated value to `ProductFormSection.SettingsRow`'s `price` and `inventory` cases so that the tap action is only handled by `ProductFormViewController` when it's editable
- In `ProductFormActionsFactory`, returned the actions for the expected editable/readonly fields for non-core products
- In `ProductDetailsFactory`, updated so that all non-simple products (including non-core products) are editable when the Products switch is on
- Added/updated unit tests

## Testing

Prerequisite: the store has at least one non-core product (not simple/external/variable/grouped)

### Products switch is on
- Go to Settings > Experimental Features, and turn on the Products switch
- Go to the Products tab
- Tap on a non-core product (not simple/external/variable/grouped) --> it should be editable with the following fields:
  - Name
  - Description
  - Images
  - Price settings (readonly) - only visible when a price is set
  - Inventory (readonly)
  - Short description
  - Categories
  - Tags
- Try editing some of the editable fields, and tap "Update" --> the product should be updated remotely

### Products switch is on
- Go to Settings > Experimental Features, and turn off the Products switch
- Go to the Products tab
- Tap on a non-core product (not simple/external/variable/grouped) --> the old readonly product details should be shown

## Example screenshots

non-core product without a price (the readonly price row is hidden) | non-core product with a price
-- | --
Simulator Screen Shot - iPhone 11 - 2020-08-24 at 17 33 32 | Simulator Screen Shot - iPhone 11 - 2020-08-24 at 17 33 55

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by Garance91540 — APPROVED** [review]
Waw this is awesome!

The only note I'd make is to make sure the Product Type displayed (ie "Booking") starts with an uppercase.

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormActionsFactory.swift [hunk 1]
```diff
@@ -24,6 +24,9 @@ enum ProductFormEditAction {
     case variationName
     case noPriceWarning
     case status
+    // Non-core products only
```
Comment (pmusolino) [inline WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormActionsFactory.swift]: For making clear what are the non-core products types.
```suggestion
    // Non-core products only, es. products with subscriptions
```

**Review by pmusolino — APPROVED** [review]
Looks good to me! 💯 

Code ✅ 
Testing ✅ I only noticed that when there are no reviews, the cell is still tappable. 

tap subscription product

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormActionsFactory.swift (hunk 1 above)
Comment (jaclync) [reply WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormActionsFactory.swift]: good idea! added in https://github.com/woocommerce/woocommerce-ios/pull/2694/commits/611a385d629495efbec9bf7752c1ef84259831f8

**Comment by jaclync** [comment]
Thanks for the review G and Paolo, some updates for another look!

> The only note I'd make is to make sure the Product Type displayed (ie "Booking") starts with an uppercase.

@Garance91540 yea we can capitalize the custom product type, with screenshots below. Custom product types with more than one word are trickier, because the API only returns the **slug** (e.g. `variable-subscription`). I updated to replace the dash with space and then capitalize the custom product type slug in https://github.com/woocommerce/woocommerce-ios/pull/2694/commits/9d036e4214cf88ac955cc09905184f905e6179d9 and https://github.com/woocommerce/woocommerce-ios/pull/2694/commits/4a48f7164c52521ffcf7279253a3d49aa82ed3b6 - lemme know if you see any potential issues reformatting the custom product types @pmusolino

Subscription | Booking | Variable subscription
-- | -- | --
Simulator Screen Shot - iPhone 11 - 2020-08-25 at 09 09 34 | Simulator Screen Shot - iPhone 11 - 2020-08-25 at 09 09 39 | Simulator Screen Shot - iPhone 11 - 2020-08-25 at 09 22 41

> I only noticed that when there are no reviews, the cell is still tappable.

@pmusolino I fixed this in https://github.com/woocommerce/woocommerce-ios/pull/2694/commits/7d9242e22addaa478a37a0908f4f9b9b35322e4d 🙂

**Comment by pmusolino** [comment]
@jaclync 👏 

> I updated to replace the dash with space and then capitalize the custom product type slug in 9d036e4 and 4a48f71 - lemme know if you see any potential issues reformatting the custom product types @pmusolino

Everything looks good and I don't see any issue at the moment. 👍

**Pull request merged**